### ISSUE TITLE
[no-shadow] flags "is already declared in the upper scope" for enum declaration

### ISSUE THREAD

**Issue by NicoleRauch** [typescript-eslint/typescript-eslint#2779]
- [ ] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

Linting a TypeScript file that contains only the following line
```
export enum Asefsadfasdf { X }
```
flags  the error  "'Asefsadfasdf' is already declared in the upper scope".
(You can even remove the export and the error remains, but of course you will also get an additional "unused" error.)

And yes, nothing else in my codebase uses this name...

```JSON
{
  "rules": {
        "no-shadow": [
            "error",
            {
                "hoist": "all"
            }
        ],
  }
}
```

```TS
export enum Asefsadfasdf { X }
```

**Expected Result**

No error on a unique enum.

**Actual Result**

The unique enum is flagged with "is already declared in the upper scope".

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.8.1` |
| `@typescript-eslint/parser`        | `4.8.1` |
| `TypeScript`                       | `3.8.3` |
| `ESLint`                           | `7.13.0` |
| `node`                             | `12.18.1` |

**Comment by bradzacher** [comment]
https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md#i-am-using-a-rule-from-eslint-core-and-it-doesnt-work-correctly-with-typescript-code

You ticked the box saying you read the FAQ...

**Comment by scottfavre** [comment]
If everyone has to make this change to allow exporting of enums (seems like a common thing to do), why is it not in the default config?

**Comment by bradzacher** [comment]
> If everyone has to make this change to allow exporting of enums

The issue is not with the code at all. Nothing to do with enums being exported.

The issue is with the eslint config.
Read the FAQ article - it explains how to correctly configure your code.

> why is it not in the default config?

`no-shadow` is not a recommended rule in the eslint set, so we do not recommend it in our configs either.
If a user decides to manually turns on `no-shadow`, there's nothing we can do - they have to manually configure our extension rule which understands TS.

**Comment by scottfavre** [comment]
is there a way to export an enum without violating the `no-shadow` rule?

**Comment by bradzacher** [comment]
> The issue is not with the code at all. ***Nothing to do with enums being exported.***

> A user has to manually configure our extension rule which understands TS.

Please read the above FAQ article so you can reconfigure your ESLint appropriately.
